compass_app/app/integration_test/app_server_data_test.dart

```diff
@@ -0,0 +1,207 @@
+import 'dart:io';
+
+import 'package:compass_app/config/dependencies.dart';
+import 'package:compass_app/main.dart';
+import 'package:compass_app/ui/activities/widgets/activities_screen.dart';
+import 'package:compass_app/ui/auth/login/widgets/login_screen.dart';
+import 'package:compass_app/ui/auth/logout/widgets/logout_button.dart';
+import 'package:compass_app/ui/booking/widgets/booking_screen.dart';
+import 'package:compass_app/ui/core/ui/custom_checkbox.dart';
+import 'package:compass_app/ui/core/ui/home_button.dart';
+import 'package:compass_app/ui/home/widgets/home_screen.dart';
+import 'package:compass_app/ui/results/widgets/result_card.dart';
+import 'package:compass_app/ui/results/widgets/results_screen.dart';
+import 'package:compass_app/ui/search_form/widgets/search_form_screen.dart';
+import 'package:flutter/foundation.dart';
+import 'package:flutter_test/flutter_test.dart';
+import 'package:integration_test/integration_test.dart';
+import 'package:provider/provider.dart';
+import 'package:shared_preferences/shared_preferences.dart';
+
+/// This Integration Test starts the Dart server
+/// before launching the Compass-App with the remote configuration.
+/// The app connects to its endpoints to perform login and create a booking.
+void main() {
+  IntegrationTestWidgetsFlutterBinding.ensureInitialized();
+
+  group('end-to-end test with remote data', () {
+    const port = '8080';
```

Comment:
Consider a comment indicating how you decided on this port and what else would need to change if the value changed.

Reply:
8080 is the default port as set by the Dart shelf template. Both the app and the server are configured to use it by default.

It shouldn't be necessary to specify the port in this e2e integration test, so this probably is due to copying the `Process.start(...` section from the server tests.

@ericwindmill fyi, the port can be set both as env variable when launching the server, and when configuring the `ApiClient` in the app side. This could be documented in the root `README.md`.

Reply:
I'll add this documentation to the README, which I'm working on now